### PR TITLE
[cex] fix trade amount for sell trades, added editOrder

### PR DESCRIPTION
**Pull request by npomfret**

### CODE REVIEW

File: js/cex.js [hunk 1]
```diff
@@ -783,15 +783,41 @@ module.exports = class cex extends Exchange {
...
                 const tradeTimestamp = this.parse8601 (tradeTime);
                 const tradeAmount = this.safeFloat (item, 'amount');
                 const tradePrice = this.safeFloat (item, 'price');
+                const feeCost = this.safeFloat (item, 'fee_amount');
                 let absTradeAmount = tradeAmount < 0 ? -tradeAmount : tradeAmount;
                 let tradeCost = undefined;
                 if (tradeSide === 'sell') {
                     tradeCost = absTradeAmount;
-                    absTradeAmount = tradeCost / tradePrice;
+                    absTradeAmount = (feeCost + tradeCost) / tradePrice;
```
Comment (kroitor) [inline js/cex.js]: This is very strange, the fee is not the part of `cost = amount * price` formula... Can you elaborate, plz?

Comment (npomfret) [reply js/cex.js]: The CEX api is very confusing.

For sell orders, the `amount` of the trade needs to be calculated (as it's not provided).  So we use the `price` and the `cost` to reverse engineer the `amount`.  I noticed from my records though that the calculated figure wasn't quite right for sell orders.  It seems that the value provided for the trade cost was net of the trade fee.  I committed an example as a comment, here is the working: 

The order was for exactly 0.002BTC, and fully executed in a single trade, so...

The trade cost is: 16.7GBP
The trade price is: 8365
So the calculated amount was: `16.7 / 8365` = *0.001996413628*BTC (close, but not correct).

If you add the fee to the cost, it becomes `(16.7 + 0.03) / 8365` = *0.002*BTC (which is exactly correct).

Comment (npomfret) [reply js/cex.js]: @kroitor does that make sense?

Comment (kroitor) [reply js/cex.js]: @npomfret yes, it does, and it means that we are not calculating the proper amount of the trade in this case when parsing trades for this exchange.

Think this way: we have an order for 0.002BTC, that has been closed in a single trade.

In that case all users would expect the following to be true:
- order['amount'] = order['trades'][0]['amount']
- order['fee'] = order['trades'][0]['fee']
- order['average'] = order['trades'][0]['price']

↑ If the above three conditions don't hold true, then the exchange is not sound in terms of financial accounting rules.

If somehow order['amount'] != order['trades'][0]['amount'], then either the left hand value or the right hand value is incorrect. You suppose that order['amount'] is incorrect in this case, but in fact you are comparing it to an incorrect trade['amount'] in the first place. So, the source trade data has to be corrected in this case, not the resulting order calculation.

Do you agree?

Comment (kroitor) [reply js/cex.js]: Ah, pardon, looks like that is exactly what you did in this PR. Sorry for the misunderstanding on my side.

Comment (kroitor) [reply js/cex.js]: We can't use the + operator, though, it has to be `this.sum`

File: js/cex.js [hunk 2]
```diff
@@ -817,7 +817,7 @@ module.exports = class cex extends Exchange {
                 let tradeCost = undefined;
                 if (tradeSide === 'sell') {
                     tradeCost = absTradeAmount;
-                    absTradeAmount = (feeCost + tradeCost) / tradePrice;
```
Comment (npomfret) [inline js/cex.js]: @kroitor thanks - keep forgetting this one!

**Pull request merged**